### PR TITLE
Add a histogram producer

### PR DESCRIPTION
**Pull request by Barthelemy**
- Add the producer for histograms
- Send a TObjArray from the histograms producer.

### CODE REVIEW

**Comment by Barthelemy** [comment]
Let me know if you think it is better to have a separate runHistoProducer.

File: Framework/include/QualityControl/HistoProducer.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+/// \brief Returns a random data producer specification which publishes on {"TST", "RAWDATA", <index>}
+///
+/// \param myParam  The value the producer should produce.
+/// \return         A fixed number producer specification
+framework::DataProcessorSpec getHistoProducerSpec(size_t myParam);
+
+/// \brief Returns an algorithm generating random messages
+///
+/// \param output   Origin, Description and SubSpecification of data to be produced
+/// \param  myParam  The value the producer should produce.
```
Comment (knopers8) [inline Framework/include/QualityControl/HistoProducer.h:33]: The description of `myParam` is unclear in this context (same in the function above)

File: Framework/src/HistoProducer.cxx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+///
+/// \file   HistoProducer.cxx
+/// \author Barthelemy von Haller
+///
+#include "QualityControl/HistoProducer.h"
+
+#include <random>
+#include <Common/Timer.h>
+#include <TH1F.h>
+#include <Common/Timer.h>
```
Comment (knopers8) [inline Framework/src/HistoProducer.cxx:20]: Timer included twice

**Comment by Barthelemy** [comment]
So I have split it. 

To be noted that the try...catch in the Printer, to test if it is a TObjArray, is not very nice. The solution Giulio advised is to use `TypeIdHelpers::uniqueId<T>()` from `Framework/TypeIdHelpers.h`, put it in an extra header and use the same in a switch on the receiving side. I will give it a try later on, but for this simple utility it is good enough I believe.

**Comment by knopers8** [comment]
Thanks. I am not merging it in case that you want to improve that try-catch thing before.

**Comment by Barthelemy** [comment]
I think that it is ok for now like that.

**Pull request merged**